testing/engine.go

```diff
@@ -180,7 +181,9 @@ func (t *Test) RunPhases() error {
 		return err
 	}
 
-	t.transaction.ProcessLogging()
+	// The tests check the values in the transaction so we can't close it yet.
```

Comment:
Realized for this test specifically merging doesn't work, but if I'm not mistaken a user would never have that issue so this could still be reasonable, let me know any thoughts.